sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageCrc64Calculator.java

```diff
@@ -0,0 +1,2579 @@
+            uCrc = updateCrcWithTables(uCrc);
+
+            uCrc ^= ByteBuffer.wrap(src, pData + 24, 8).order(ByteOrder.LITTLE_ENDIAN).getLong() ^ uCrc3;
+            uCrc = updateCrcWithTables(uCrc);
+
+            pData += 32;
+        }
+
+        // Process remaining bytes
+        for (uBytes = 0; uBytes < uSize; ++uBytes, ++pData) {
+            uCrc = (uCrc >>> 8) ^ M_U1[(int) ((uCrc ^ src[pData]) & 0xFF)];
+        }
+
+        return uCrc ^ ~0L; // Final bit flip
+    }
+
+    // Helper method to update CRC using lookup tables
+    private static long updateCrcWithTables(long uCrc) {
+        for (int i = 0; i < 8; i++) {
+            uCrc = (uCrc >>> 8) ^ M_U1[(int) (uCrc & 0xFF)];
+        }
+        return uCrc;
+    }
+
+    /**
+     * Concatenates two CRC64 values by combining their initial and final CRC values and sizes.
+     * This method ensures unsigned behavior and uses the `mulX_N` method to perform necessary
+     * multiplications in GF(2^64).
+     *
+     * @param uInitialCrcAB The initial CRC value for the concatenated data.
+     * @param uInitialCrcA The initial CRC value for the first data segment.
+     * @param uFinalCrcA The final CRC value for the first data segment.
+     * @param uSizeA The size of the first data segment.
+     * @param uInitialCrcB The initial CRC value for the second data segment.
+     * @param uFinalCrcB The final CRC value for the second data segment.
+     * @param uSizeB The size of the second data segment.
+     * @return The concatenated CRC64 value.
+     */
+    public static long concat(long uInitialCrcAB, long uInitialCrcA, long uFinalCrcA, long uSizeA, long uInitialCrcB,
+        long uFinalCrcB, long uSizeB) {
+        long uFinalCrcAB = uFinalCrcA ^ ~0L; // Invert bits for unsigned logic
+
+        // Ensure unsigned behavior when comparing uInitialCrcA and uInitialCrcAB
+        if ((uInitialCrcA & 0xFFFFFFFFFFFFFFFFL) != (uInitialCrcAB & 0xFFFFFFFFFFFFFFFFL)) {
+            // Apply mulX_N with proper unsigned masking
+            uFinalCrcAB ^= multiplyCrcByPowerOfX((uInitialCrcA ^ uInitialCrcAB) & 0xFFFFFFFFFFFFFFFFL, uSizeA)
+                & 0xFFFFFFFFFFFFFFFFL;
+        }
+
+        uFinalCrcAB ^= (uInitialCrcB ^ ~0L) & 0xFFFFFFFFFFFFFFFFL; // Ensure unsigned XOR logic
+        uFinalCrcAB = multiplyCrcByPowerOfX(uFinalCrcAB & 0xFFFFFFFFFFFFFFFFL, uSizeB) & 0xFFFFFFFFFFFFFFFFL;
+        uFinalCrcAB ^= uFinalCrcB & 0xFFFFFFFFFFFFFFFFL;
+
+        return uFinalCrcAB & 0xFFFFFFFFFFFFFFFFL; // Ensure the result is treated as unsigned
+    }
+
+    /**
+     * Multiplies a CRC value by x^n in GF(2^64).
+     * This method uses a lookup table to perform the multiplication efficiently.
+     *
+     * @param a The CRC value to be multiplied.
+     * @param uSize The power of x by which the CRC value is to be multiplied.
+     * @return The resulting CRC64 value after multiplication.
+     */
+    public static long multiplyCrcByPowerOfX(long a, long uSize) {
```

Comment:
For this method and `mulPolyUnrolled` we should probably make them private as from the SDK standpoint we will only be using the compute and concat entrypoints.